File: 1299/main.go
package main 

import (
	"fmt"
)

func main() {
	fmt.Println(replaceElements([]int{17,18,5,4,6,1}))
}

func replaceElements(arr []int) []int {
	fm := func (nums []int) int {
		mx := nums[0]
		for i := 1; i < len(nums); i++ {
			if nums[i] > mx {
				mx = nums[i]
			}
		}
		return mx
	}
	m := fm(arr)
	arr[0] = m

	for i := 0; i < len(arr)-1; i++ {
		if m == arr[i] {
			m = fm(arr[i+1:])
		}
		arr[i] = m
	}

	arr[len(arr)-1] = -1
	return arr
}

File: 1299/main.rs
struct Solution {}

impl Solution {
    pub fn replace_elements(arr: Vec<i32>) -> Vec<i32> {
        let n = arr.len();
        let mut ans = vec![0; n];
        ans[n-1] = -1;
        for i in (0..n-1).rev() {
            ans[i] = ans[i+1].max(arr[i+1]);
        }
        ans
    }
}

fn main() {
    println!("{:#?}", Solution::replace_elements(vec![17,18,5,4,6,1]));
}